### PR TITLE
0.9.0

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
joserfc 0.9.0

**Destination channel:** defaults

### Links

- [PKG-4469](https://anaconda.atlassian.net/browse/PKG-4469) 
- [Upstream repository](https://github.com/authlib/joserfc)
- [Upstream changelog/diff](https://github.com/authlib/joserfc/releases/tag/0.9.0)
- Relevant dependency PRs:
  - moto

### Explanation of changes:

- get into main with lint changes 

[PKG-4469]: https://anaconda.atlassian.net/browse/PKG-4469?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JamesRobertsonGames** [comment]
I have assumed we decrease build number when this is our first build but it is build number 1 on forge. Happy for someone to correct me 👍

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,33 +10,43 @@ source:
...
-    - python >=3.8
+    - python
     - setuptools
     - pip
+    - wheel
   run:
-    - python >=3.8
+    - python
     - cryptography
+    - pycryptodome
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: `pycryptodome` is an optional dependency, see https://github.com/authlib/joserfc/blob/2470d72dec9aa15c316325aafc35ca41ec638f66/pyproject.toml#L30-L31.
It's not mentioned in the source code but docs https://github.com/search?q=repo%3Aauthlib%2Fjoserfc%20pycryptodome&type=code

Only `cryptography` is required https://github.com/authlib/joserfc/blob/b669af5fbf17ad500893004299debe006885dd97/pyproject.toml#L5-L7

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,33 +10,43 @@ source:
...
+    - pytest
 
 about:
   home: https://jose.authlib.org/en/dev/
-  summary: The ultimate Python library for JOSE RFCs, including JWS, JWE, JWK, JWA, JWT
+  summary: Implementations of JOSE RFCs in Python
+  description: |
+    joserfc is a Python library that provides a comprehensive implementation of several essential JSON Object Signing and Encryption (JOSE) standards.
   dev_url: https://github.com/authlib/joserfc
+  doc_url: https://jose.authlib.org/en/dev/
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://jose.authlib.org
```
Minor, but this is friendlier to non-english users.

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
pycryptodome is optional: https://github.com/authlib/joserfc/blob/0.9.0/pyproject.toml#L31

**Comment by skupr-anaconda** [comment]
> I have assumed we decrease build number when this is our first build but it is build number 1 on forge. Happy for someone to correct me 👍

That's fine to reset the build number

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**